pkg/datapath/linux/routing/info.go

```diff
@@ -58,11 +61,11 @@ func (info *RoutingInfo) GetInterfaceNumber() int {
 // (on either ENI or Azure interface) is the only supported path currently.
 // Azure does not support masquerade yet (subnets CIDRs aren't provided):
 // until it does, we forward a masquerade bool to opt out ipam.Cidrs use.
-func NewRoutingInfo(gateway string, cidrs []string, mac, ifaceNum string, masquerade bool) (*RoutingInfo, error) {
-	return parse(gateway, cidrs, mac, ifaceNum, masquerade)
+func NewRoutingInfo(gateway string, cidrs []string, mac, ifaceNum, ipamMode string, masquerade bool) (*RoutingInfo, error) {
+	return parse(gateway, cidrs, mac, ifaceNum, ipamMode, masquerade)
 }
 
-func parse(gateway string, cidrs []string, macAddr, ifaceNum string, masquerade bool) (*RoutingInfo, error) {
+func parse(gateway string, cidrs []string, macAddr, ifaceNum, ipamMode string, masquerade bool) (*RoutingInfo, error) {
```

Comment:
do we need to update the below as well ? 

https://github.com/cilium/cilium/blob/224ef2045ed3cc44a16b366aae55cfd231fac770/pkg/datapath/linux/routing/routing_test.go#L248-L253

Reply:
I agree, but how is the test currently passing :thinking:
```
$ TESTPKGS="pkg/datapath/linux/routing" sudo -E make tests-privileged
[...]
START: routing_test.go:31: LinuxRoutingSuite.TestConfigure
PASS: routing_test.go:31: LinuxRoutingSuite.TestConfigure	0.046s

START: routing_test.go:49: LinuxRoutingSuite.TestConfigureRoutewithIncompatibleIP
level=warning msg="Unable to configure rules and routes because IP is not an IPv4 address" endpointIP="fd00::2" subsys=linux-routing
PASS: routing_test.go:49: LinuxRoutingSuite.TestConfigureRoutewithIncompatibleIP	0.002s

START: routing_test.go:66: LinuxRoutingSuite.TestDelete
Test: valid IP addr matching rules
Test: IP addr doesn't match rules
level=warning msg="No rule matching found" rule="20: from all to 192.168.2.233/32 lookup main" subsys=linux-routing
Test: IP addr matches more than number expected
level=warning msg="Found too many rules matching, skipping deletion" candidates="[ip rule 20: from all to 192.168.2.233/32 table 254 ip rule 20: from 192.168.2.123/32 to 192.168.2.233/32 table 254]" rule="20: from all to 192.168.2.233/32 lookup main" subsys=linux-routing
PASS: routing_test.go:66: LinuxRoutingSuite.TestDelete	0.058s

START: routing_test.go:58: LinuxRoutingSuite.TestDeleteRoutewithIncompatibleIP
level=warning msg="Unable to delete rules because IP is not an IPv4 address" endpointIP="fd00::2" subsys=linux-routing
PASS: routing_test.go:58: LinuxRoutingSuite.TestDeleteRoutewithIncompatibleIP	0.000s

OK: 9 passed
--- PASS: Test (0.15s)
PASS
[...]
```

I must be missing something obvious here...

Reply:
it's weird, below is my fresh build in vagrant runtime box

```
vagrant@runtime1:~/go/src/github.com/cilium/cilium$ git show -s --format=%h
5e191a6407
vagrant@runtime1:~/go/src/github.com/cilium/cilium$ TESTPKGS="pkg/datapath/linux/routing" sudo -E make tests-privileged
make init-coverage
make[1]: Entering directory '/home/vagrant/go/src/github.com/cilium/cilium'
echo "mode: count" > coverage-all-tmp.out
echo "mode: count" > coverage.out
make[1]: Leaving directory '/home/vagrant/go/src/github.com/cilium/cilium'
for pkg in github.com/cilium/cilium/pkg/datapath/linux/routing; do \
	PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/vagrant/go/src/github.com/cilium/cilium/bpf go test  -mod=vendor -tags=osusergo,privileged_tests -ldflags "-X github.com/cilium/cilium/pkg/kvstore.consulDummyAddress=https://consul:8443 -X github.com/cilium/cilium/pkg/kvstore.etcdDummyAddress=http://etcd:4002 -X github.com/cilium/cilium/pkg/datapath.DatapathSHA256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" $pkg -test.v -timeout 600s -check.vv  -covermode=count -coverprofile=coverage.out -coverpkg github.com/cilium/cilium/pkg/datapath/linux/routing \
	|| exit 1; \
	tail -n +2 coverage.out >> coverage-all-tmp.out; \
done
# github.com/cilium/cilium/pkg/datapath/linux/routing [github.com/cilium/cilium/pkg/datapath/linux/routing.test]
pkg/datapath/linux/routing/routing_test.go:248:31: not enough arguments in call to parse
	have (string, []string, string, string, bool)
	want (string, []string, string, string, string, bool)
FAIL	github.com/cilium/cilium/pkg/datapath/linux/routing [build failed]
FAIL
make: *** [Makefile:138: tests-privileged] Error 1
```

Reply:
With Tam's help, I found the cause for the always-passing test and the fix is at https://github.com/cilium/cilium/pull/19231.

I've also fixed the compilation issue in this PR.